docs/state/substates.md

```diff
@@ -5,63 +5,109 @@ from pcweb.templates.docpage import docdemo_from
 ```
 
 # Substates
-As your app grows, your state will grow too.  You can split your state into
-multiple substates from your base state to keep things organized.
+A substate is a class that inherits from another state class, providing a means for two state classes to share common state resources, such as variables or event handlers.
 
 ## Creating a Substate
-Your base state should inherit from `rx.State`. Substates can either inherit
-from your base state or other substates.
+A substate can typically inherit from a base state(a state that inherits from `rx.State`) or another substate.
+Let’s take a look at a simple example illustrating how and when to use substates. 
+The code below shows two states, `RectangleState` and `SquareState`, each representing metadata about rectangles and squares, along with the calculation of their respective areas.
 
 ```python exec
-class ParentState(State):
-    checked: bool = True
-    count: int = 0
+class RectangleState(rx.State):
+    name: str = "rectangle"
+    shape_type: str = "quadrilateral"
+    length: int = 10
+    width: int = 5
 
+    @rx.var
+    def area(self) -> int:
+        return self.length * self.width 
 
-class ChildState1(ParentState):
-    value: int = 42
+class SquareState(rx.State):
+    name: str = "square"
+    shape_type: str = "quadrilateral"
+    length: int = 4
 
+    @rx.var
+    def area(self) -> int:
+        return self.length ** 2
 
-class ChildState2(ParentState):
-    color: str = "red"
-
-
-class ChildState3(ChildState1):
-    text: str = "Hello World"
 
 
 def my_badge():
-    return rx.badge(ChildState3.text, color_scheme=ChildState2.color)
+    return rx.vstack(
+        rx.hstack(
+            rx.text(RectangleState.name),
+            rx.text(RectangleState.area),
+        ),
+        rx.hstack(
+            rx.text(SquareState.name),
+            rx.text(SquareState.area),
+        ),
+    )
 ```
 
 ```python eval
 docdemo_from(
-    ParentState,
-    ChildState1,
-    ChildState2,
-    ChildState3,
+    RectangleState,
+    SquareState,
     component=my_badge,
 )
 ```
 
-In the example above, we have a base state, `ParentState`, with two substates
-`ChildState1` and `ChildState2`. Additionally, `ChildState3` inherits from
-`ChildState1`. Components can access any var or event handler from any substate.
 
-A common use case may be to create a substate for each page of your app, while
-keeping general vars such as the logged in user in the base state for easy
-access.
+
+In this code, you can see that both states share a common resource (`shape_type`). We can improve code organization and promote reusability by introducing the concept of substates. 
+This involves creating a base state (`QuadrilateralState`), from which other states can inherit or subclass. After refactoring, the code takes on the following structure:
+
+```python
+class QuadrilateralState(rx.State):
```

Comment:
The quadrilateral example is a pretty simple one. If there's a better suggestion/example, would be happy to use that